src/test/resources/refaster/PreConditionsVerifierRecipes.java

```diff
@@ -0,0 +1,403 @@
+
+        /**
+         * Instantiates a new instance.
+         */
+        public UsesTypeMapWhenBeforeTemplateContainsMapRecipe() {}
+
+        @Override
+        public String getDisplayName() {
+            return "Refaster template `PreConditionsVerifier.UsesTypeMapWhenBeforeTemplateContainsMap`";
+        }
+
+        @Override
+        public String getDescription() {
+            return "Recipe created for the following Refaster template:\n```java\npublic static class UsesTypeMapWhenBeforeTemplateContainsMap {\n    \n    @BeforeTemplate()\n    void withGeneric(Map<?, ?> actual) {\n        System.out.println(actual);\n    }\n    \n    @BeforeTemplate()\n    void withGenericTwo(Map<?, ?> actual) {\n        System.out.println(actual);\n    }\n    \n    @AfterTemplate()\n    void withoutGeneric(Map actual) {\n        System.out.println(\"Changed: \" + actual);\n    }\n}\n```\n.";
+        }
+
+        @Override
+        public TreeVisitor<?, ExecutionContext> getVisitor() {
+            JavaVisitor<ExecutionContext> javaVisitor = new AbstractRefasterJavaVisitor() {
+                final JavaTemplate withGeneric = JavaTemplate
+                        .builder("System.out.println(#{actual:any(java.util.Map<?,?>)});")
+                        .build();
+                final JavaTemplate withGenericTwo = JavaTemplate
+                        .builder("System.out.println(#{actual:any(java.util.Map<?,?>)});")
+                        .build();
+                final JavaTemplate withoutGeneric = JavaTemplate
+                        .builder("System.out.println(\"Changed: \" + #{actual:any(java.util.Map)});")
+                        .build();
+
+                @Override
+                public J visitMethodInvocation(J.MethodInvocation elem, ExecutionContext ctx) {
+                    JavaTemplate.Matcher matcher;
+                    if ((matcher = withGenericTwo.matcher(getCursor())).find()) {
+                        return embed(
+                                withoutGeneric.apply(getCursor(), elem.getCoordinates().replace(), matcher.parameter(0)),
+                                getCursor(),
+                                ctx,
+                                SHORTEN_NAMES
+                        );
+                    }
+                    if ((matcher = withGeneric.matcher(getCursor())).find()) {
+                        return embed(
+                                withoutGeneric.apply(getCursor(), elem.getCoordinates().replace(), matcher.parameter(0)),
+                                getCursor(),
+                                ctx,
+                                SHORTEN_NAMES
+                        );
+                    }
+                    return super.visitMethodInvocation(elem, ctx);
+                }
+
+            };
+            return Preconditions.check(
+                    Preconditions.and(
+                            new UsesType<>("java.util.Map", true),
```

Comment:
Name: **UsesTypeMapWhenBeforeTemplateContainsMapOnlyRecipe**, thus:
- Use `new UsesType<>("java.util.Map", true)`

Reply:
In this case indeed I would not expect any change to the preconditions.